lightning/src/ln/onion_utils.rs

```diff
@@ -1712,6 +1712,49 @@ impl Into<LocalHTLCFailureReason> for u16 {
 	}
 }
 
+impl_writeable_tlv_based_enum!(LocalHTLCFailureReason,
```

Comment:
Perhaps some explanation in the commit message or code why we don't just keep writing a u16 code?